crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3681,6 +3727,8 @@ pub(crate) mod tests {
     #[test_case(Ty::BuiltinClassLiteral("str"), Ty::SubclassOfAny)]
     #[test_case(Ty::AbcInstance("ABCMeta"), Ty::SubclassOfBuiltinClass("type"))]
     #[test_case(Ty::SubclassOfBuiltinClass("str"), Ty::BuiltinClassLiteral("str"))]
+    #[test_case(Ty::IntLiteral(1), Ty::AlwaysFalsy)]
+    #[test_case(Ty::IntLiteral(0), Ty::AlwaysTruthy)]
```

Comment:
I think it would be good to also test here that an instance of some builtin type (`str` is fine) is not a subtype of either `AlwaysTruthy` or `AlwaysFalsy`.